libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/RouteViewModel.java

```diff
@@ -94,15 +94,13 @@ public void fetchRouteNewOrigin(Point newOrigin) {
    */
   private void fetchRoute(Point origin, Point destination) {
     if (origin != null && destination != null) {
-      NavigationRoute.Builder routeBuilder = NavigationRoute.builder()
-        .accessToken(Mapbox.getAccessToken());
+      Double bearing
+        = rawLocation.hasBearing() ? Float.valueOf(rawLocation.getBearing()).doubleValue() : null;
 
-      if (locationHasBearing()) {
-        fetchRouteWithBearing(routeBuilder, origin);
-      } else {
-        fetchRouteWithoutBearing(routeBuilder, origin);
-      }
-      routeBuilder.destination(destination);
+      NavigationRoute.builder()
+        .accessToken(Mapbox.getAccessToken())
+        .origin(origin, bearing, 90d)
```

Comment:
Will adding a `null` bearing but still including a 90 degree tolerance have any effect on the request?

Reply:
Shouldn't be an issue since in the directions SDK we have a check:
```java
if(angle != null && tolerance != null) {
  this.bearings.add(new Double[]{angle, tolerance});
} else {
  this.bearings.add(new Double[0]);
}
```

if either value is null, we essentially skip the bearing value in the URL with a simple `;` which is what is specified in the spec.

Reply:
Sweet, looks good to go then 🚗